src/unit/test_util.c

```diff
@@ -0,0 +1,299 @@
+    UNUSED(flags);
+
+    char buf[32];
+    long double v;
+    int sz;
+
+    v = 0.0 / 0.0;
+    sz = ld2string(buf, sizeof(buf), v, LD_STR_AUTO);
+    TEST_ASSERT(sz == 3);
+    TEST_ASSERT(!strcmp(buf, "nan"));
+
+    return 0;
+}
+
+int test_fixedpoint_d2string(int argc, char **argv, int flags) {
+    UNUSED(argc);
+    UNUSED(argv);
+    UNUSED(flags);
+
+    char buf[32];
+    double v;
+    int sz;
+    v = 0.0;
+    sz = fixedpoint_d2string(buf, sizeof buf, v, 4);
+    TEST_ASSERT(sz == 6);
+    TEST_ASSERT(!strcmp(buf, "0.0000"));
+    sz = fixedpoint_d2string(buf, sizeof buf, v, 1);
+    TEST_ASSERT(sz == 3);
+    TEST_ASSERT(!strcmp(buf, "0.0"));
+    /* set junk in buffer */
+    memset(buf,'A',32);
+    v = 0.0001;
+    sz = fixedpoint_d2string(buf, sizeof buf, v, 4);
+    TEST_ASSERT(sz == 6);
+    TEST_ASSERT(buf[sz] == '\0');
+    TEST_ASSERT(!strcmp(buf, "0.0001"));
+    /* set junk in buffer */
+    memset(buf,'A',32);
+    v = 6.0642951598391699e-05;
+    sz = fixedpoint_d2string(buf, sizeof buf, v, 4);
+    TEST_ASSERT(sz == 6);
+    TEST_ASSERT(buf[sz] == '\0');
+    TEST_ASSERT(!strcmp(buf, "0.0001"));
+    v = 0.01;
+    sz = fixedpoint_d2string(buf, sizeof buf, v, 4);
+    TEST_ASSERT(sz == 6);
+    TEST_ASSERT(!strcmp(buf, "0.0100"));
+    sz = fixedpoint_d2string(buf, sizeof buf, v, 1);
+    TEST_ASSERT(sz == 3);
+    TEST_ASSERT(!strcmp(buf, "0.0"));
+    v = -0.01;
+    sz = fixedpoint_d2string(buf, sizeof buf, v, 4);
+    TEST_ASSERT(sz == 7);
+    TEST_ASSERT(!strcmp(buf, "-0.0100"));
+     v = -0.1;
+    sz = fixedpoint_d2string(buf, sizeof buf, v, 1);
+    TEST_ASSERT(sz == 4);
+    TEST_ASSERT(!strcmp(buf, "-0.1"));
+    v = 0.1;
+    sz = fixedpoint_d2string(buf, sizeof buf, v, 1);
+    TEST_ASSERT(sz == 3);
+    TEST_ASSERT(!strcmp(buf, "0.1"));
+    v = 0.01;
+    sz = fixedpoint_d2string(buf, sizeof buf, v, 17);
+    TEST_ASSERT(sz == 19);
+    TEST_ASSERT(!strcmp(buf, "0.01000000000000000"));
+    v = 10.01;
+    sz = fixedpoint_d2string(buf, sizeof buf, v, 4);
+    TEST_ASSERT(sz == 7);
+    TEST_ASSERT(!strcmp(buf, "10.0100"));
+    /* negative tests */
+    sz = fixedpoint_d2string(buf, sizeof buf, v, 18);
+    TEST_ASSERT(sz == 0);
+    sz = fixedpoint_d2string(buf, sizeof buf, v, 0);
+    TEST_ASSERT(sz == 0);
+    sz = fixedpoint_d2string(buf, 1, v, 1);
+    TEST_ASSERT(sz == 0);
+
+    return 0;
+}
+
+#if defined(__linux__)
+/* Since fadvise and mincore is only supported in specific platforms like
+ * Linux, we only verify the fadvise mechanism works in Linux */
+static int cache_exist(int fd) {
+    unsigned char flag;
+    void *m = mmap(NULL, 4096, PROT_READ, MAP_SHARED, fd, 0);
+    TEST_ASSERT(m);
+    TEST_ASSERT(mincore(m, 4096, &flag) == 0);
+    munmap(m, 4096);
+    /* the least significant bit of the byte will be set if the corresponding
+     * page is currently resident in memory */
+    return flag&1;
+}
+
+int test_reclaimFilePageCache(int argc, char **argv, int flags) {
```

Comment:
I think we need to move this declaration outside of the ifdef, since we won't be able to build the tests on non-linux machines since it will complain it's not defined.

Reply:
Okay, I will take a look.